src/thor/trace_route_action.cc

```diff
@@ -126,13 +126,13 @@ void thor_worker_t::route_match(Api& request) {
   // TODO - make sure the trace has timestamps..
   auto& options = *request.mutable_options();
   bool use_timestamps = options.use_timestamps();
-  std::vector<PathInfo> path_infos;
+  m_path_infos.clear();
```

Comment:
of course because of this refactoring we have a lot of calls to clear like this which happen at the time of the subsequent request. we have a method called `thor_worker_t::cleanup` which should be used for these clears. the reason for that is because then we dont spend time while answering the actual request on clearing the data structures. this pattern allows us to remove this latency from the actual request servicing. can you please move all `clear` calls to the cleanup function.

Reply:
question: is there any other dependent data structures that the cleanup() function clears?
If so, will clearing those affect this function?

Reply:
There are intermediate data structures only serves for a function call as a temp storage, and there are data structures lives longer than that function call. If so, those clear() can not be called at the same time. Maybe I should group the intermediate data structures together to do a separate clear(). It may eventually falls on the design pattern.